event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/ChannelAllocationState.java

```diff
@@ -53,22 +59,31 @@ public synchronized SlabRef detach() {
         return slab;
     }
 
-    public synchronized SlabAllocation allocate(int desired) {
-        checkArgument(desired > 0);
+    public synchronized SlabAllocation allocate(PeekingIterator<Integer> eventSizes) {
+        checkArgument(eventSizes.hasNext());
 
         if (!isAttached()) {
             return null;
         }
 
+        int offset = _slabConsumed;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "offset". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "offset" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1481)